src/components/Form/index.tsx

```diff
@@ -0,0 +1,86 @@
+import RjsForm from '@rjsf/mui';
+import ajvValidator from '@rjsf/validator-ajv8';
+import { FormProps as RjsFormProps } from '@rjsf/core';
+import { useCallback } from 'react';
+import { Box, Button, ButtonProps } from '@mui/material';
+import { PasswordWidget } from './PasswordWidget';
+import { TextWidget } from './TextWidget';
+import { TextareaWidget } from './TextareaWidget';
+import { SelectWidget } from './SelectWidget';
+import { DateTimeWidget } from './DateTimeWidget';
+import { UpDownWidget } from './UpDownWidget';
+
+const internalWidgets: {
+	[k: string]: any;
+} = {
+	PasswordWidget,
+	EmailWidget: TextWidget,
+	TextWidget,
+	TextareaWidget,
+	SelectWidget,
+	UpDownWidget,
+	DateTimeWidget,
+};
+
+export interface MuiFormProps extends Omit<RjsFormProps, 'validator'> {
+	/** If true, do not display the form submit button */
+	hideSubmitButton?: boolean;
+	/** Properties that are passed to the submit button, these are the same props used for the [`Button`](#button) component */
+	submitButtonProps?: ButtonProps;
+	/** If passed, it will show a secondary button, these are the same props used for the [`Button`](#button) component */
+	secondaryButtonProps?: ButtonProps;
+
+	validator?: RjsFormProps['validator'];
+}
+
+export const MuiForm: React.FC<React.PropsWithChildren<MuiFormProps>> = ({
+	hideSubmitButton,
+	submitButtonProps,
+	secondaryButtonProps,
+	validator = ajvValidator,
+	widgets,
+	fields,
+	children,
+	...otherProps
+}) => {
+	const handleKeyUpDown = useCallback(
+		(e: React.KeyboardEvent<HTMLDivElement>) => {
+			if (e.key === 'Enter' && !hideSubmitButton) {
+				e.stopPropagation();
+			}
+		},
+		[hideSubmitButton],
+	);
+	return (
+		<Box onKeyDown={handleKeyUpDown} onKeyUp={handleKeyUpDown}>
+			<RjsForm
+				validator={validator}
+				showErrorList={false}
+				onChange={(data) => {
+					otherProps.onChange?.(data);
+				}}
+				widgets={{ ...internalWidgets, ...(widgets || {}) }}
+				{...otherProps}
+			>
+				{hideSubmitButton && <span />}
+
+				{!hideSubmitButton && (
+					<Button
+						{...submitButtonProps}
+						color="customBlue"
+						variant="contained"
+						type="submit"
+					/>
+				)}
```

Comment:
We may need to talk about this a bit, cc @drskullster 
I know this is what we already had in rendition but if you have it this way then the form's `onSubmit` does not fire with the `enter` button when the submit button is hidden (since it is technically not rendered). It's better to have `display: 'none'` when hidden IMO. However, I believe Jon noted that this may cause issues with accessibility. We should find a way forward that enables usage of native browser behavior (i.e. submit form on enter)

Reply:
It was already in my TODO list to discuss this point today during the call. This is exactly why I've added the children property, as all form actions should be handled inside the Form.

Reply:
@myarmolinsky @JSReds we can talk about it in our call but the TLDR; is that I'm ok with this implementation, because we can have external submit buttons very easily:

```html
<form id="my-form">
  <input type="text" />
</form>
<p>
  <button form="my-form" type="submit">External submit</button> <!-- works! -->
</p>
```

I'm wondering if we could make without the `<span/>` element though, any idea why this would be required?

Reply:
But the user shouldn't have to use an external submit button, right? Shouldn't we just have the proper form support from the getgo?